pkg/statsd/handler_http_forwarder_v2.go

```diff
@@ -134,6 +144,14 @@ func NewHttpForwarderHandlerV2(
 	if concurrentMerge <= 0 {
 		return nil, fmt.Errorf("concurrent-merge must be positive")
 	}
+	compressionType, err := web.ReadCompressionType(compressionTypeStr)
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
```

Comment:
You can simplify this by returning `err` here, `fmt.Errorf` is more for additional written context to help understand the issue. For example:

```go
return nil, fmt.Errorf("unable to read compression: %w")
```